pycjs/open_digger.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Python 3.9
+
+# @Time   : 2024/1/14 10:26
+# @Author : 'Lou Zehua'
+# @File   : pycjs_node_vm.py
+
+import copy
+import pandas as pd
+
+from node_vm2 import NodeVM
+
+from pycjs.pycjs_node_vm import get_export_module
+
+
+vm_option_open_digger = {
+    # 'console': 'inherit',
+    # "wrapper": 'commonjs',
+    # 'sandbox': {},
+    'require': {
+        'external': '../lib',
+        # 'builtin': ['os'],
+        # 'import': ["ijavascript-plotly"],
+        'root': '../',
+        # 'mock': { }
+    }
+}
+
+js_import_open_digger = """
+var openDigger = require('../src/open_digger.js');
+"""
+
+if __name__ == '__main__':
```

Comment:
What does this mean? Is this an unit test of this file?

Reply:
Yes, it is. This an unit test of this file.
The code segment under the `if __name__ == '__main__':` statement will be executed only when the `.py` file is executed as the unique entrance for main:0 thread of a python application, which means it cannot be executed when the module is initialized from an import statement.
In this typical case, it will be executed only when the `open_digger.py` is directly executed: 
1. Run `python ./pycjs/open_digger.py`,
2. Run `python -m open_digger` in the pycjs folder, because relative module names is not supported.

---
Related Introduction and Experiments: 
The file `__init__.py` has the same output as any typical `.py` file when it comes to the `if __name__ == '__main__':` statement.
Special situation: the `__main__.py` file in a module.
```python
print("__main__.py is executed.")
if __name__ == '__main__':
    print("__main__.py __name__ is __main__!")
```
If the code above exists in `pycjs/__main__. py` and a module init file `pycjs/__init__. py` is exist, there will be 3 ways to execute it:
[√] 1. Run `python ./pycjs/__main__.py`
[√] 2. Run `python ./pycjs`, it will choose the `__main.py` in `./pycjs` as the unique entrance.
[√] 3. Run `python -m pycjs` in the pycjs folder as relative module names is not supported.

However, it is not supported in this way: run `python -m __main__` in the pycjs folder. I still cannot understand the case as it report an error about module specification: `Error while finding module specification for '__main__' (ValueError: __main__.__spec__ is None)` .